src/runtimes/node/native_modules/plugin.js

```diff
@@ -0,0 +1,68 @@
+const path = require('path')
+
+const readPackageJson = require('read-package-json-fast')
+
+const { isNativeModule } = require('./detector')
+
+const packageName = /^([^@][^/]*|@[^/]*\/[^/]+)(?:\/|$)/
+
+const findNativeModule = (packageJsonPath, cache) => {
+  if (cache[packageJsonPath] === undefined) {
+    // eslint-disable-next-line fp/no-mutation, no-param-reassign, promise/prefer-await-to-then
+    cache[packageJsonPath] = readPackageJson(packageJsonPath).then(
+      (data) => Boolean(isNativeModule(data)),
+      () => {},
+    )
+  }
+
+  return cache[packageJsonPath]
+}
+
+const externalNativeModulesPlugin = (externalizedModules) => ({
+  name: 'external-native-modules',
+  setup(build) {
+    const cache = {}
+
+    // eslint-disable-next-line complexity, max-statements
+    build.onResolve({ filter: /^([^./]*)$/ }, async (args) => {
```

Comment:
Would merging the `packageName` regExp to `filter` improve performance?

Reply:
From my tests, no. I think this is because the regex is more complex, which gets more expensive since it'll be executed for each import. Also, the regex we're currently using for filtering already filters out relative and absolute paths, so the number of callback invokations that we'd reduce by using `packageName` would be minimal.

Reply:
And we'd still need to run `packageName.exec` inside the callback, because we need to extract the directory from the package name.

Reply:
:+1: